### ISSUE TITLE
[ISSUE] databricks_job email_notifications field is always reconfigured

### ISSUE THREAD

**Issue by Gnarik** [databrickslabs/terraform-provider-databricks#324]
Hi there,

When I create a databricks job with emails in the email_notifications field the job is properly created but when I rerun the same exact strict, terraform tries to re-setup the email_notifications field. Which cause a job to be redeployed unexpectedly and can cause running job to fail.

### Terraform Version
```
Your version of Terraform is out of date! The latest version
is 0.13.3. You can update by downloading from https://www.terraform.io/downloads.html
Terraform v0.13.0
+ provider registry.terraform.io/databrickslabs/databricks v0.2.4
+ provider registry.terraform.io/hashicorp/aws v3.6.0
```

### Affected Resource(s)
Please list the resources as a list, for example:
- databricks_job

### Terraform Configuration Files
```hcl
terraform {
  required_providers {
    databricks = {
      source = "databrickslabs/databricks"
      version = "0.2.4"
    }
  }
}

provider "aws" {
  region  = "us-east-1"
}

provider "databricks" {
  host = "https://ENDPOINT/"
  token = var.databricks_api_token
}
resource "databricks_job" "job" {
  library {
    jar = "s3://${data.aws_s3_bucket_object.job_jar.bucket}/${data.aws_s3_bucket_object.job_jar.key}"
  }
  name = "test-job"
  new_cluster {
    autoscale {
      min_workers = 6
      max_workers = 15
    }
    spark_version = "6.4.x-scala2.11"
    aws_attributes {
      zone_id = var.aws_region
      availability = "SPOT"
      spot_bid_price_percent = 100
      instance_profile_arn = var.databricks_instance_profile_arn
      first_on_demand = 1
      ebs_volume_type = "GENERAL_PURPOSE_SSD"
      ebs_volume_count = 1
      ebs_volume_size = 32
    }
    node_type_id = "r4.2xlarge"
  }


  email_notifications {
    on_failure = ["rom.par@men.tier"]
  }

  timeout_seconds = 3600
  max_retries = 1
  max_concurrent_runs = 100
  jar_main_class_name = "com.mycompany.service.Main


}
data "aws_s3_bucket_object" "job_jar" {
  bucket = "BUCKET_NAME"
  key    = "data_ingestion/job:latest.jar"
}
```

### Debug Output
https://gist.github.com/Gnarik/937c5c4f5d9a2e41d2eade065c8a646a < 1st apply
https://gist.github.com/Gnarik/ca4cb3d7ad185b3324e1dfbb36eb3e39 < 2nd apply

### Expected Behavior
During the 2nd apply nothing should be done on the job.

### Actual Behavior
Terraform tries to change some fields on the job.

### Steps to Reproduce
1. `terraform apply`
2. `terraform apply`

**Comment by nfx** [comment]
@Gnarik 

1. `jar_main_class_name` is deprecated and will be removed in 0.3, there was a respectable warning message during the plan. Please rewrite the code the way there are no warnings and try again. This is what might be causing the issue. I've checked internally and `POST ../jobs/reset` should not cause a running job to fail. 
2. you're using 0.2.4 and the latest version of provider is 0.2.5

**Comment by Gnarik** [comment]
Hi,

Thank you for your answer. 

I'm sorry I've disrespected the respectable warning message. Here is the new code with a new gist. There is still WARN messages but I'm not sure I can do something to get rid of them.

```hcl
terraform {
  required_providers {
    databricks = {
      source = "databrickslabs/databricks"
      version = "0.2.5"
    }
  }
}

provider "aws" {
  region  = "us-east-1"
}

provider "databricks" {
  host = "https://ENDPOINT/"
  token = var.databricks_api_token
}
resource "databricks_job" "job" {
  library {
    jar = "s3://${data.aws_s3_bucket_object.job_jar.bucket}/${data.aws_s3_bucket_object.job_jar.key}"
  }
  name = "test-job"
  new_cluster {
    autoscale {
      min_workers = 6
      max_workers = 15
    }
    spark_version = "6.4.x-scala2.11"
    aws_attributes {
      zone_id = "us-east-1a"
      availability = "SPOT"
      spot_bid_price_percent = 100
      instance_profile_arn = "arn:aws:iam::ACCOUNT_ID:instance-profile/identity-dev-databricks-mws-role"
      first_on_demand = 1
      ebs_volume_type = "GENERAL_PURPOSE_SSD"
      ebs_volume_count = 1
      ebs_volume_size = 32
    }
    node_type_id = "r4.2xlarge"
  }


  email_notifications {
    on_failure = ["rom.par@men.tier"]
  }

  timeout_seconds = 3600
  max_retries = 1
  max_concurrent_runs = 100

  spark_jar_task {
    main_class_name = "com.mycompany.service.Main"
  }


}
data "aws_s3_bucket_object" "job_jar" {
  bucket = "BUCKET_NAME"
  key    = "data_ingestion/job:latest.jar"
}
```
https://gist.github.com/Gnarik/9ef8cffdf12dd9a105b838e24b7d1dae

> I've checked internally and POST ../jobs/reset should not cause a running job to fail.

Ok, thank you. I thought it was the cause of my issue, I'll dig elsewhere.

**Comment by Gnarik** [comment]
Hi, I manually changed it in order not to share it on github but the email address we use on our system are valid addresses.
We are also investigating regard `I've checked internally and POST ../jobs/reset should not cause a running job to fail.` as we keep having this issue and the only thing that appears in the plan logs is this email change. I'll come back with more information when we have been able to dig into it but if you have furthermore information, they are welcome.

**Comment by nfx** [comment]
@Gnarik  Verified that changing job while it was running did not affect the current job run (made a 15-minute sleep notebook & changed spark_config & email_notifications few times). Most likely email_notification was reconfigured because invalid values supplied. It might be an error with this specific job. 

upcoming 0.2.6 might contain a fix for it - #334

**Comment by nfx** [comment]
@Gnarik released in 0.2.7

**Comment by romaric-sticky** [comment]
Hi, thank you for letting me know.
